src/ec2/utils.ts

```diff
@@ -4,13 +4,27 @@
  */
 
 import { EC2 } from 'aws-sdk'
-import { AsyncCollection } from "../shared/utilities/asyncCollection"
+import { AsyncCollection } from '../shared/utilities/asyncCollection'
+import globals from '../shared/extensionGlobals'
+import { pageableToCollection } from '../shared/utilities/collectionUtils'
 
-export function extractInstanceIdsFromReservations(reservations: AsyncCollection<EC2.ReservationList | undefined>): AsyncCollection<string> {
+export function extractInstanceIdsFromReservations(
```

Comment:
this could live in the "client". Do we have that for ec2 yet? That's the usual pattern we have done thus far. (We have some improvements planned, but until then it probably makes sense to continue the pattern. For ec2 that would mean creating a client in `src/shared/clients/ec2Client.ts`